publicsuffixlist.js

```diff
@@ -645,3 +645,40 @@ class PublicSuffixList {
 export default new PublicSuffixList();
 
 /******************************************************************************/
+
+const psURL = new URL('ws://x');
+
+// Experimental alternative to punycode.js but only for entries in the public
+// suffix list.
+
+export function publicSuffixToASCII(publicSuffix) {
```

Comment:
Hopefully the comments are good enough but I'm happy to explain.

From what I understand so far only the `*` is a problem. If otherwise the public suffix fails to convert, it means it's an error in the public suffix list. We're supposed to check for such errors and ignore such entries anyway.